### PR TITLE
Adds support for tournament accounts and models

### PR DESCRIPTION
**Pull request by pschork**
This change incorporates updates to the Numerai tournament API in anticipation of the rollout of a new account system with multi-model support - see Background below for more details.
 
Adds get_account to return private account information and deprecates get_user
Replaces all internal calls to get_user with calls to get_account
Adds get_account_transactions to return NMR wallet transactions and deprecates get_user_transactions
Adds get_models convenience method to return model name -> id mapping
Updates get_submission_filenames to accept optional model_id for accounts w/multiple models
Updates get_user to accept optional model_id for accounts w/multiple models
Updates get_payments to accept optional model_id for accounts w/multiple models
Updates get_transactions to accept optional model_id for accounts w/multiple models
Updates get_stakes to accept optional model_id for accounts w/multiple models
Updates upload_predictions to accept optional model_id for accounts w/multiple models
Updates stake_change to accept optional model_id for accounts w/multiple models
Updates stake_drain to accept optional model_id for accounts w/multiple models
Updates stake_decrease to accept optional model_id for accounts w/multiple models
Updates stake_increase to accept optional model_id for accounts w/multiple models
Adds tests for get_account, get_account_transactions and get_models
Updates Cli for all endpoints that accept model_id
Update README.md

**Background**
The legacy Numerai Tournament data model coupled user profile, authentication, wallet, and tournament submissions into a singular user model. Users were allowed to create 3 accounts (recently increased to 10) to allow for model experimentation. 
```
┌─────────────────┐  ┌─────────────────┐  ┌─────────────────┐
│     USER 1      │  │     USER 2      │  │     USER 3      │
│ ┌─────────────┐ │  │ ┌─────────────┐ │  │ ┌─────────────┐ │
│ │ AUTH        │ │  │ │ AUTH        │ │  │ │ AUTH        │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ WALLET      │ │  │ │ WALLET      │ │  │ │ WALLET      │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ SUBMISSIONS │ │  │ │ SUBMISSIONS │ │  │ │ SUBMISSIONS │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ STAKES      │ │  │ │ STAKES      │ │  │ │ STAKES      │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ SCORES      │ │  │ │ SCORES      │ │  │ │ SCORES      │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ ...         │ │  │ │ ...         │ │  │ │ ...         │ │
│ └─────────────┘ │  │ └─────────────┘ │  │ └─────────────┘ │
└─────────────────┘  └─────────────────┘  └─────────────────┘
```                                                             
Over time, users asked for a more streamlined auth system so that they did not have to manage mulitple emails, password, api keys, wallets, etc. As we were considering the re-launch of Numerai Forums including SSO support, we began migrating our auth & wallet subsystems under a new account model. To maintain backwards compatibility, we shimmed out user APIs to be backwards compatible and maintained a 1:1 mapping of accounts:users.                                                              
```                                                             
┌─────────────────┐  ┌─────────────────┐  ┌─────────────────┐
│   ACCOUNT 1     │  │   ACCOUNT 2     │  │   ACCOUNT 3     │
│ ┌─────────────┐ │  │ ┌─────────────┐ │  │ ┌─────────────┐ │
│ │ AUTH        │ │  │ │ AUTH        │ │  │ │ AUTH        │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ WALLET      │ │  │ │ WALLET      │ │  │ │ WALLET      │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ MODELS      │ │  │ │ MODELS      │ │  │ │ MODELS      │ │
│ └──────┬──────┘ │  │ └──────┬──────┘ │  │ └──────┬──────┘ │
└────────┼────────┘  └────────┼────────┘  └────────┼────────┘
         │                    │                    │         
         │                    │                    │         
         ▼                    ▼                    ▼         
┌─────────────────┐  ┌─────────────────┐  ┌─────────────────┐
│     USER 1      │  │     USER 2      │  │     USER 3      │
│ ┌─────────────┐ │  │ ┌─────────────┐ │  │ ┌─────────────┐ │
│ │ SUBMISSIONS │ │  │ │ SUBMISSIONS │ │  │ │ SUBMISSIONS │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ STAKES      │ │  │ │ STAKES      │ │  │ │ STAKES      │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ SCORES      │ │  │ │ SCORES      │ │  │ │ SCORES      │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ ...         │ │  │ │ ...         │ │  │ │ ...         │ │
│ └─────────────┘ │  │ └─────────────┘ │  │ └─────────────┘ │
└─────────────────┘  └─────────────────┘  └─────────────────┘
```                                                             
We are now ready to expose the new Account and Model (formerly Users) APIs externally. While we are still maintain backwards compatibility for 1:1 mappings, in order to allow for Accounts to hold multiple models we have updated out APIs to require model UUID to be specified when an Account holds > 1 model.                                                            
```
┌─────────────────┐  ┌─────────────────┐  ┌─────────────────┐
│   ACCOUNT 1     │  │   ACCOUNT 2     │  │   ACCOUNT 3     │
│                 │  │ ┌─────────────┐ │  │                 │
│  !TOMBSTONED!   │  │ │ AUTH        │ │  │  !TOMBSTONED!   │
│                 │  │ ├─────────────┤ │  │                 │
│     ░░░░░░      │  │ │ WALLET      │ │  │     ░░░░░░      │
│    ░░RIP░░░     │  │ ├─────────────┤ │  │    ░░RIP░░░     │
│    ░░░░░░░░     │  │ │ MODELS      │ │  │    ░░░░░░░░     │
│    ░░░░░░░░     │  │ └──────┬──────┘ │  │    ░░░░░░░░     │
└─────────────────┘  └────────┼────────┘  └─────────────────┘
                              │                              
                              │                              
         ┌────────────────────┼────────────────────┐         
         │                    │                    │         
         │                    │                    │         
         ▼                    ▼                    ▼         
┌─────────────────┐  ┌─────────────────┐  ┌─────────────────┐
│  USER/MODEL 1   │  │  USER/MODEL 2   │  │  USER/MODEL 3   │
│ ┌─────────────┐ │  │ ┌─────────────┐ │  │ ┌─────────────┐ │
│ │ SUBMISSIONS │ │  │ │ SUBMISSIONS │ │  │ │ SUBMISSIONS │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ STAKES      │ │  │ │ STAKES      │ │  │ │ STAKES      │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ SCORES      │ │  │ │ SCORES      │ │  │ │ SCORES      │ │
│ ├─────────────┤ │  │ ├─────────────┤ │  │ ├─────────────┤ │
│ │ ...         │ │  │ │ ...         │ │  │ │ ...         │ │
│ └─────────────┘ │  │ └─────────────┘ │  │ └─────────────┘ │
└─────────────────┘  └─────────────────┘  └─────────────────┘
```
The ability for end users to absorb/import existing models into a primary Account is in early testing phase. Before we can launch we need third-party APIs to update with new model_id requirements - hence this PR.

### CODE REVIEW

File: numerapi/numerapi.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
 # -*- coding: utf-8 -*-
 
+from __future__ import absolute_import
```
Comment (uuazed) [inline numerapi/numerapi.py]: I've removed python2 support recently and removed that future import. Any reason for re-introducing it or did you just branch of before the removal?

File: numerapi/numerapi.py [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
+
 # System
 import zipfile
 import os
 import logging
 import datetime
 import decimal
-from typing import List, Dict
```
Comment (uuazed) [inline numerapi/numerapi.py]: I added type hints on purpose, any reason for removing them?

**Review by uuazed — COMMENTED** [review]
In general it looks good. Thanks for paying attention to all the little details (readme update, tests, CLI changes etc)!

Your PR reverts many of the recent changes in the master branch. Could you please do a rebase?

File: numerapi/numerapi.py (hunk 2 above)
Comment (pschork) [reply numerapi/numerapi.py]: not intended

File: numerapi/numerapi.py (hunk 1 above)
Comment (pschork) [reply numerapi/numerapi.py]: not intended

**Pull request merged**